### DISCUSSION TITLE
Strategies for incorporating elemental data in EBSD analysis with MTEX

### DISCUSSION THREAD

**Discussion by nyyssont** [mtex-toolbox/mtex#1244] [Feature Requests]
Hello,

I noticed from a discussion that it is possible to import h5oina files. In those, it is possible to incorporate EDS data in the EBSD measurement. Is this data also imported into MTEX?

Here are some possible uses for combining EDS and EBSD analyses:
1. Separate different crystallographically similar phases from each other.
2. Determine average chemical compositions for each grain to study e.g. segregation.
3. Determine compositional differences at grain boundaries vs. grain interiors.

Please update this list if you can think of additional items!

Also, I would be particularly interested to hear from any experiences of these types of analyses you may have done, or thought you would like to do with MTEX. All other discussion on the topic is also welcome.

-Tuomo

**Comment by kilir** [comment]
Hi,
yes EDS data is also imported and written into the ebsd variable. h5oina files should work, however since I only have a limited set of test files available, it would actually be nice if someone could try the function and report on any errors.

One pitfall is that the hdf5 standard in Matlab isn't en par with hdf5 v1.10 so one needs to use hdf5format_convert (https://github.com/HDFGroup/hdf5) before being able to read the file.

Cheers,
Rüdiger

**Reply by nyyssont** [reply]
I tried out the function to load a h5oina file, which resulted in a fatal error and Matlab closing. I'll try to convert the file and report if I have success.

**Reply by nyyssont** [reply]
The function appears to work using Matlab 2021b, if there are no commas in field names. When I tried to load a map with a field for Bromine, I got the following error message:

>> loadEBSD_h5oina('Mamta.h5oina')
Invalid field name: 'Br_La1,2'.

Error in loadEBSD_h5oina (line 92)
            EDSdata.(sane_name)=double(h5read(fname,[EDS_data.Name '/' EDS_data.Datasets(thing).Name]));

Note: there is probably no bromine in the actual map. It had been assigned by the AutoID feature in Aztec. Removing Br and re-exporting to h5oina allowed to load the map.

**Reply by kilir** [reply]
Hi Tuomo,

Can you try if this a9143f61bcdf072011d41170b7ff073b32f6d989 also allows to use more funny names with a comma?

Support for hdf5 v1.10 has been requested since 2018, so good to know that it finally arrived. For those using earlier versions of Matlab, still something like `h5format_convert -v some_input.h5oina` needs to be used.

Cheers,
Rüdiger

**Reply by nyyssont** [reply]
Thanks Rüdiger,

the fix allows to load the file with Br_La1,2 and switches the comma to an underline.

-Tuomo

**Comment by lisakemp** [comment]
Rudiger and Tuomo –

Regarding HDF5 support:

As it turns out that 1.10 support in the MATLAB HDF functions just came out in MATLAB R2021b<https://www.mathworks.com/help/matlab/release-notes.html?rntext=HDF5&startrelease=R2021b&endrelease=R2021b&groupby=release&sortby=descending&searchHighlight=HDF5>, a couple of weeks ago.

Does this address some of the extra effort required to handle the HDF5 files for EBSD data in MTEX?

I’d be interested to hear, @Tuomo, if the h5oina is related or a separate issue.

Thanks,
Lisa
Lisa Kempler
MATLAB Community Strategist, MathWorks
508-647-7462
www.mathworks.com/products<http://www.mathworks.com/products/matlab>
www.mathworks.com/solutions/earth-ocean-atmospheric-sciences.html<https://www.mathworks.com/solutions/earth-ocean-atmospheric-sciences.html>
https://serc.carleton.edu/teaching_computation/index.html<https://serc.carleton.edu/teaching_computation/index.html>

**Reply by nyyssont** [reply]
Hello Lisa,

thank you for this information. I upgraded to 2021b and I could then load a h5oina file using `loadEBSD_h5oina`.
